src/node_embedding_api.h

```diff
@@ -0,0 +1,23 @@
+//
+// Description: C-based API for embedding Node.js.
+//
+// !!! WARNING !!! WARNING !!! WARNING !!!
+// This is a new API and is subject to change.
+// While it is C-based, it is not ABI safe yet.
+// Consider all functions and data structures as experimental.
+// !!! WARNING !!! WARNING !!! WARNING !!!
+//
+// This file contains the C-based API for embedding Node.js in a host
+// application. The API is designed to be used by applications that want to
+// embed Node.js as a shared library (.so or .dll) and can interop with
+// C-based API.
+//
+
+#include "node.h"
+
+#ifdef _WIN32
```

Comment:
This header file must follow the design patterns used by Node-API code:
- It must not depend on the `node.h`. The `node.h` is a non-ABI stable C++ API. It should rather depend on the `node-api.h` or `js_native_api.h`.
- For the `__cdecl` there is a macro `NAPI_CDECL`. It must help to avoid the code duplication.
- We must use the `EXTERN_C_START` and `EXTERN_C_END` macros around the declaration of functions to enable that header to be used from C++.